src/twisted/spread/test/test_pb.py

```diff
@@ -170,6 +169,52 @@ def disconnectClientFactory():
 
 
 
+@implementer(interfaces.IConnector)
+class ReconnectingConnector(object):
+    """
+    An L{IConnector} implementation that calls the specified callback
+    when its C{connect} method is called
+
+    @param reconnectCallback: the callback to call when C{connect} is
+        called
+    @type reconnectCallback: A callable
+
+    @param *args: Position arguments to pass C{reconnectCallback}
+    @param **kwargs: Keyword arguments to pass to C{reconnectCallback}
+    """
+
+    def __init__(self, reconnectCallback, *args, **kwargs):
+        self.reconnectCallback = reconnectCallback
```

Comment:
As per https://glyph.twistedmatrix.com/2015/05/separate-your-fakes-and-your-inspectors.html we should not expose any public attributes on the test versions of stuff that are not present on the "real" versions of stuff.  These should all be `_`-prefixed.